yeonjae/programmers-2/이진변환반복하기.js

```diff
@@ -0,0 +1,18 @@
+function solution(s) {
+  const result = [0, 0];
+
+  while (s !== "1") {
+    s = s.split("");
+
+    let temp = s.filter((el) => el === "1").length;
+    result[0]++;
+    result[1] += s.length - temp;
+    s = temp.toString(2);
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(s) {
  const result = [0, 0];

  while (s !== "1") {
    const deletedZeroStrLength = s.split("").filter((el) => el === "1").length;
    
    result[0]++;
    result[1] += s.length - deletedZeroStrLength;
    s = temp.toString(2);
  }

  return result;
}
```
- 풀이잘봤습니다~ 🙇‍♂️
- temp는 let이 아닌 `const`로 선언해주세요~ 또한 temp보다는 deletedZeroStrLength(?) 와 같이 좀 더 디테일한 네이밍도 괜찮을 것 같다는 생각이 들었습니다~
- 근데 근본적으로 매번 반복문 할 때마다 `문자열을 -> 배열로 변환하고 -> 배열을 다시 문자열로 변환 후 할당`하는 구조가 조금 비효율적이라고 느꼈어요
- 아래와 같이 replace를 활용한 풀이를 참고해주세요~
```js
function solution(s) {
  const result = [0, 0];

  while (s.length > 1) {
    const convertedStr = s.replace(/0/g, () => {
      result[1]++; // 제거된 0의 개수 카운팅
      return "";
    });

    s = convertedStr.length.toString(2);
    result[0]++; // 변환 횟수 카운팅
  }

  return result;
}
```

Reply:
replace쓰니까 진짜 훨씬깔끔하네요,,,, temp부분을 놓쳤네요ㅠㅠㅠ 코멘트 감사합니다!